From: =?utf-8?q?Lukas_M=C3=A4rdian?= <slyon@ubuntu.com>
Date: Wed, 12 Mar 2025 11:51:44 +0100
Subject: hw/s390x/sclp.c: include s390-virtio-ccw.h to make S390_CCW_MACHINE
 available
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

Partial backport of "3d9836e4 s390x/sclp: Simplify get_sclp_device()"

Needed for LP#2049698 backport, to avoid FTBFS:
../../hw/s390x/sclp.c: In function ‘read_SCP_info’:
../../hw/s390x/sclp.c:172:39: error: implicit declaration of function ‘S390_CCW_MACHINE’ [-Wimplicit-function-declaration]
  172 |     s390_ipl_convert_loadparm((char *)S390_CCW_MACHINE(machine)->loadparm,
      |                                       ^~~~~~~~~~~~~~~~
../../hw/s390x/sclp.c:172:39: warning: nested extern declaration of ‘S390_CCW_MACHINE’ [-Wnested-externs]
../../hw/s390x/sclp.c:172:64: error: invalid type argument of ‘->’ (have ‘int’)
  172 |     s390_ipl_convert_loadparm((char *)S390_CCW_MACHINE(machine)->loadparm,
      |                                                                ^~

Origin: backport, https://gitlab.com/qemu-project/qemu/-/commit/3d9836e4
Bug-Ubuntu: https://launchpad.net/bug/2049698
Last-Update: 2025-03-12
---
 hw/s390x/sclp.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hw/s390x/sclp.c b/hw/s390x/sclp.c
index be6bbb3..873b126 100644
--- a/hw/s390x/sclp.c
+++ b/hw/s390x/sclp.c
@@ -21,6 +21,7 @@
 #include "hw/s390x/s390-pci-bus.h"
 #include "hw/s390x/ipl.h"
 #include "hw/s390x/cpu-topology.h"
+#include "hw/s390x/s390-virtio-ccw.h"
 
 static inline SCLPDevice *get_sclp_device(void)
 {
